### ISSUE TITLE
[v2.9] Update Shepherd Tag for 1.29 Support

### ISSUE THREAD

**Issue by krunalhinguu** [rancher/rancher#45564]
## Objective
To update the shepherd tag in the rancher/rancher repository for the v2.9 branch as it is a release blocker as part of 1.29 support.
ref: https://github.com/rancher/rancher/blob/a8e15f2678302f646bf6b835ff213e9b3338424e/go.mod#L5

## Steps to Complete the Task

- [ ] **Update rancher/shepherd Repository:**

   - Verify that the rancher/apis and other relevant tags are updated in the rancher/shepherd repository.

- [ ] **Update rancher/rancher Repository:**

   - Once above step is completed and changes are merged in the rancher/shepherds repository, utilize the latest commit from that PR.
   - Update the shepherd tag in the rancher/rancher repository for the v2.9 branch.

- [ ]  **Raise Pull Request in rancher/rancher Repository:**

   - After updating the shepherd tag in the rancher/rancher repository, raise a pull request (PR).
   - Ensure the changes are reviewed and get the PR merged into the v2.9 branch.

cc: @snasovich @kinarashah

**Comment by igomez06** [comment]
Hi @krunalhinguu this isn't exactly what we agreed on, in our meeting last Monday.

We’re not going to merge Ugur's fork to Shepherd, there’s no need for that at the moment, what we’re waiting right now is info from you that all the forked rancher dependency versions you had in your go.mod are merged.

1. We will update Ugur's fork (current Shepherd PR) without merging to Shepherd, in his fork’s go mod, the only dependency is going to be your pkg/api fork version.
2. Then we'll provide you a new forked Shepherd tag, and at that point it’s reviewing and merging 1.29 PR (not quite sure when)
3. Once 1.29 PR is in r/r then we’ll update the same PR for r/r/pkg/apis.  Do the same PR steps again this time without any fork versions, merge it to Shepherd. Merge the PR at this step.
4. Update r/r to the new Shepherd official version instead of my fork.

**Comment by krunalhinguu** [comment]
@igomez06 I think that's what these issue is about. These are the steps we are going to follow once r/r is merged and we will have final version for r/r/apis

**Comment by igomez06** [comment]
@krunalhinguu so this is the ticket tracking after the initial r/r PR is merged?

**Comment by krunalhinguu** [comment]
@igomez06 that is correct. Once r/r will be merged in release/2.9 we will follow this steps.

Created now because while reviewing r/r PR, it is creating some confusion. So i have created these issue for their better clarity.

Also most probably i will be owning step 2 and step 3.

You guys just have to update and merge shepherd PR.

**Comment by igomez06** [comment]
Sounds good thank you Krunal! Sorry for the misunderstanding!

**Comment by caliskanugur** [comment]
@krunalhinguu thank you! To get approvals and merge the Shepherd PR, a Rancher PR is sometimes provided to ensure everything is fine on the r/r side. If everyone agrees, I can take step 2. and 3. as it's just going to be updating the existing linked PR https://github.com/rancher/rancher/pull/45631

**Comment by krunalhinguu** [comment]
cc - @mitulshah-suse

**Comment by mitulshah-suse** [comment]
Closing the issue since testing is not necessarily needed for this.
